src/nemo_evaluator/environments/gym.py

```
@@ -396,56 +408,83 @@ def _build_cmd(self) -> list[str] | str:
-        while time.monotonic() < deadline:
+        while True:
+            remaining = deadline - time.monotonic()
+            if remaining <= 0:
+                break
             if self._process.poll() is not None:
                 output = ""
                 if self._process.stdout:
                     output = self._process.stdout.read().decode(errors="replace")
                 raise RuntimeError(
                     f"Server exited with code {self._process.returncode} during startup.\nOutput:\n{output}"
                 )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)